server/src/main/java/com/ncc/neon/services/ClusterService.java

```diff
@@ -328,4 +375,53 @@ private String convertNumberToTextBin(BigDecimal number) {
     public void setClusterClause(ClusterClause clusterClause) {
         this.clusterClause = clusterClause;
     }
+
+    private FieldType retrieveFieldType(String type) {
+        switch (type) {
+            case "boolean":
+                return FieldType.BOOLEAN;
+            case "byte":
+            case "integer":
+            case "long":
+            case "short":
+                return FieldType.INTEGER;
+            case "date":
+                return FieldType.DATETIME;
+            case "number":
+            case "double":
+            case "float":
+            case "half_float":
+            case "scaled_float":
+                return FieldType.DECIMAL;
+            case "geo-point":
+            case "geo-shape":
+                return FieldType.GEO;
+            case "keyword":
+                return FieldType.KEYWORD;
+            case "nested":
+            case "object":
+                return FieldType.OBJECT;
+            case "text":
+            default:
+                return FieldType.TEXT;
+        }
+    }
```

Comment:
So I think I was unintentionally a bit misleading here.  The `clusterClause.getFieldType()` is ALREADY a `FieldType`, and the "cluster type" is a bit different.  I'd recommend something like the following (you'd have to create the ClusterType enum):

```java
switch(type) {
  case FieldType.DECIMAL:
  case FieldType.INTEGER:
    return ClusterType.NUMBER;
  case FieldType.KEYWORD:
  case FieldType.TEXT:
    return ClusterType.STRING;
  case FieldType.DATETIME:
    return ClusterType.DATE;
  // So right now latitude and longitude are considered decimals, so we'll need a special case for the lat/lon clustering.
  // Note that "geo-point" and "geo-shape" are supposed to label specific constructs and NOT lat/lon points.
  case "latlon": // I don't remember if I used "geo" or "latlon" in the JIRA ticket... whichever one is clearer.
    return ClusterType.LAT_LON;
  default:
    return null;
}
```

Does that make sense?

Reply:
I started working on Lat/Lon and was wondering what the different between that and geo-point was and now this makes more sense! Thank you, I'll fix this and merge it into the lat/lon branch.

Reply:
I'm not sure if this is the correct way to do it, but since the fieldType comes in as a string, I can't get the actual `FieldType` enum so I can't switch on it. Furthermore, if I were to switch on the actual `FieldType` enum, I couldn't also have the `"latlon"` case. One other thing I noticed is that `FieldType` is adapter specific, so it would be necessary to figure out which adapter we're currently using so that we could get the `FieldType` that corresponds with it. 

I have it currently switching on a string so that it can switch on the value directly from the clusterclause. Please feel free to let me know if you don't prefer this behavior.

Reply:
Sorry I missed this.  Looking at your commit, the approach you used seems fine.  Thanks for changing it!